test/Dialect/FIRRTL/errors.mlir

```diff
@@ -555,7 +555,47 @@ firrtl.circuit "MemoryPortsWithDifferentTypes" {
 firrtl.circuit "SubfieldOpFieldError" {
   firrtl.module @SubfieldOpFieldError() {
     %w = firrtl.wire  : !firrtl.bundle<a: uint<2>, b: uint<2>>
-    // expected-error @+1 {{subfield element index is greater than the number of fields}}
+    // expected-error @+1 {{}}
     %w_a = firrtl.subfield %w(2) : (!firrtl.bundle<a : uint<2>, b : uint<2>>) -> !firrtl.uint<2>
   }
 }
+
+// -----
+
+firrtl.circuit "BitCast1" {
+  firrtl.module @BitCast1() {
+    %a = firrtl.wire : !firrtl.bundle<valid: uint<1>, ready: uint<1>, data: uint>
+    // expected-error @+1 {{bitwidth cannot be determined for input operand type}}
+    %b = firrtl.bitcast %a : (!firrtl.bundle<valid: uint<1>, ready: uint<1>, data: uint>) -> (!firrtl.uint<6>)
+  }
+}
+
+// -----
+
+firrtl.circuit "BitCast2" {
+  firrtl.module @BitCast2() {
+    %a = firrtl.wire : !firrtl.bundle<valid: uint<1>, ready: uint<1>, data: uint<1>>
+    // expected-error @+1 {{the bitwidth of input (3) and result (6) don't match}}
+    %b = firrtl.bitcast %a : (!firrtl.bundle<valid: uint<1>, ready: uint<1>, data: uint<1>>) -> (!firrtl.uint<6>)
+  }
+}
+
+// -----
+
+firrtl.circuit "BitCast3" {
+  firrtl.module @BitCast3() {
+    %a = firrtl.wire : !firrtl.clock
+    %b = firrtl.bitcast %a : (!firrtl.clock) -> (!firrtl.uint<1>)
```

Comment:
What causes this to error?  it looks like it should work (getWidth(clock) -> 1), and it seems reasonable for it to work.